VirtoCommerce.Storefront/Domain/Stores/StoreWorkContextBuilderExtensions.cs

```diff
@@ -22,6 +22,9 @@ public static Task WithStoresAsync(this IWorkContextBuilder builder, IEnumerable
 
             builder.WorkContext.AllStores = stores.ToArray();
             var currentStore = builder.HttpContext.GetCurrentStore(stores, defaultStoreId);
+
+            currentStore.ReplaceThemeForPreviewIfRequired(builder.HttpContext);
```

Comment:
I would like to simplify implementation and replace   all your code with this single line
`currentStore.ThemeName  = builder.WorkContext.QueryString.Get("previewtheme") ?? currentStore.ThemeName;`